lib/clickhousex/protocol.ex

```diff
@@ -250,4 +251,22 @@ defmodule Clickhousex.Protocol do
 
     {query, params}
   end
+
+  defp extract_clickhouse_url_settings(conn, opts) do
+    settings =
+      Enum.reduce(opts, [], fn {opt_name, value}, acc ->
+        opt_name = Atom.to_string(opt_name)
+
+        case opt_name do
+          "clickhouse_" <> setting ->
```

Comment:
maybe `opts[:settings]`?

Reply:
What do you mean?

Reply:
I mean that instead of prefixing the settings with `clickhouse_`, we can put them in a separate option.

Reply:
We could to that but then maybe it's good to keep the option to have it configurable also via the connection URL directly? `CLICKHOUSE_DATABASE_URL=http://127.0.0.1:8123/plausible_events_db?clickhouse_foo=1&clickhouse_bar=2`

Reply:
I like your suggestion though, I'll give it a go.

Reply:
Let me check if ecto supports that.

Reply:
Seems like it does: https://github.com/elixir-ecto/ecto/blob/03b0597ba52da2ae7245b60b32006d97323c7325/lib/ecto/repo/supervisor.ex#L88-L144, but the query string opts come flattened out.

Reply:
I'm not sure about the URL approach. In my mind, it feels a bit icky that the URL passed in config is not valid when used with clickhouse directly. Like, `curl http://127.0.0.1:8123/plausible_events_db?clickhouse_async_insert=1` is not the same as `curl http://127.0.0.1:8123/plausible_events_db?async_insert=1`

Reply:
https://github.com/plausible/clickhousex/pull/22/commits/ecaab4de2fea0112a4dd6a42729aeaa75256681e let's stick with that

Reply:
If I were doing it via the URL, I'd preserve the query string options as is.

Reply:
Yeah I was debating whether it makes sense to pass non-clickhouse query params then. E.g. we currently set the pool size via db url, which is irrelevant to clickhouse itself. So in turn it'd be probably good to filter out only the known options... meh